public/app/features/alerting/unified/components/alert-groups/AlertDetails.tsx

```diff
@@ -17,15 +17,15 @@ interface AmNotificationsAlertDetailsProps {
 
 export const AlertDetails: FC<AmNotificationsAlertDetailsProps> = ({ alert, alertManagerSourceName }) => {
   const styles = useStyles2(getStyles);
-  const isExternalAM = !isGrafanaRulesSource(alertManagerSourceName);
+  const isGrafanaAM = isGrafanaRulesSource(alertManagerSourceName);
   return (
     <>
       <div className={styles.actionsRow}>
         <Authorize
           actions={
-            isExternalAM
-              ? [AccessControlAction.AlertingInstancesExternalWrite]
-              : [AccessControlAction.AlertingInstanceCreate, AccessControlAction.AlertingInstanceUpdate]
+            isGrafanaAM
```

Comment:
I see this conditional expression is pervasive throughout the code and I wonder if we could handle that a bit differently. 
Maybe we could hide this distinction between Grafana and External alert managers.
I'm thinking about a function that gives us a set of permissions depending on the alert manager.
Something like this:
```ts
export function alertingInstancePermissions(alertManagerSourceName: string) {
  const isGrafanaAM = isGrafanaRulesSource(alertManagerSourceName);

  const amVersion = isGrafanaAM ? 'grafana' : 'external';

  const permissions = {
    read: {
      grafana: [AccessControlAction.AlertingInstanceRead],
      external: [AccessControlAction.AlertingInstancesExternalRead],
    },
    create: {
      grafana: [AccessControlAction.AlertingInstanceCreate],
      external: [AccessControlAction.AlertingInstancesExternalWrite],
    },
    update: {
      grafana: [AccessControlAction.AlertingInstanceUpdate],
      external: [AccessControlAction.AlertingInstancesExternalWrite],
    },
    delete: {
      grafana: [AccessControlAction.AlertingInstanceUpdate],
      external: [AccessControlAction.AlertingInstancesExternalWrite],
    },
  };

  return {
    read: permissions.read[amVersion],
    create: permissions.create[amVersion],
    update: permissions.update[amVersion],
    delete: permissions.delete[amVersion],
  };
}
```
And then to get list of required permissions we could use:
`const requiredPermissions = alertingInstancePermissions('alertManagerSourceName').create;`

Reply:
I like this approach and it certainly would help make the markup a lot cleaner. I'll see about creating some util functions to handle permissions